androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/state/SubscribeToPredictionsTest.kt

```diff
@@ -51,20 +49,16 @@ class SubscribeToPredictionsTest {
                     onJoin: (ApiResult<PredictionsByStopJoinResponse>) -> Unit,
                     onMessage: (ApiResult<PredictionsByStopMessageResponse>) -> Unit
                 ) {
-                    check(!isConnected) { "called connect when already connected" }
-                    isConnected = true
-                    launch { stopIdsChannel.send(stopIds) }
                     this.onJoin = onJoin
+                    launch { stopIdsChannel.send(stopIds) }
                 }
 
                 override var lastUpdated: Instant? = null
 
                 override fun shouldForgetPredictions(predictionCount: Int) = false
 
                 override fun disconnect() {
-                    check(isConnected) { "called disconnect when not connected" }
-                    isConnected = false
-                    launch { disconnectChannel.send(Unit) }
+                    /* null-op */
```

Comment:
question: Is there still a meaningful alternative way to test that disconnect is called as expected? Could we invoke `onCleared` directly in a test?

Reply:
It's looking like predictions is behaving correctly on backgrounding! I'm seeing a `SocketError` on background (after changing all the `rememberSaveable`s) but it doesn't crash. Predictions in the log start and stop as expected.

Reply:
Ugh that was a struggle but `onClear` functionality is tested now